src/contracts/core/AVSDirectory.sol

```diff
@@ -490,4 +491,381 @@ contract AVSDirectory is
+                _completeAllocation(queuedReallocations[i]);
+            } else {
+                _completeDeallocation(queuedReallocations[i]);
+            }
+        }
+    }
+
+    /**
+     * @notice Called by an AVS to slash an operator for a given operatorSetId, list of strategies, and bipsToSlash
+     * @param operator the operator to slash
+     * @param operatorSetId which operator set the operator is being slashed from
+     * @param strategies list of strategies to slash the operator for
+     * @param bipsToSlash the amount of bips to slash the operator by
+     */
+    function slashOperator(
+        address operator,
+        uint32 operatorSetId,
+        IStrategy[] calldata strategies,
+        uint32 bipsToSlash
+    ) external {
+        require(
+            0 < bipsToSlash && bipsToSlash < SlashingAccountingUtils.BIPS_FACTOR,
+            "AVSDirectory.slashOperator: invalid bipsToSlash"
+        );
+        // Slash the operator for each strategy
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            OperatorSet memory opSet = OperatorSet({avs: msg.sender, operatorSetId: operatorSetId});
+            bytes32 opSetKey = _hashStrategyAndOperatorSet(strategies[i], opSet);
+            address[] memory allocators = _operatorAllocators[operator][opSetKey];
+
+            // For each strategy, slash each of its allocators and reduce their total magnitudes
+            for (uint256 j = 0; j < allocators.length; ++j) {
+                _slashAllocator(operator, opSetKey, allocators[j], bipsToSlash);
+                emit OperatorSlashed(operator, strategies[i], opSet, allocators[j], bipsToSlash);
+            }
+        }
+    }
+
+    /**
+     *
+     *                         INTERNAL FUNCTIONS
+     *
+     */
+
+    function _queueAllocation(
+        address allocator,
+        address operator,
+        IStrategy strategy,
+        MagnitudeAdjustment[] calldata magnitudeAdjustments,
+        AllocatorOperatorDetails storage allocatorDetails
+    ) internal returns (QueuedReallocation[] memory) {
+        TotalAndNonslashableUpdate storage totalMagnitudeUpdate = _getLatestTotalAndNonslashableUpdate(operator, strategy, allocator);
+        uint64 nonslashableMagnitude = totalMagnitudeUpdate.nonslashableMagnitude;
+        uint32 nonce = allocatorDetails.reallocationNonce;
+
+        QueuedReallocation[] memory queuedAllocations = new QueuedReallocation[](magnitudeAdjustments.length);
+
+        for (uint256 i = 0; i < magnitudeAdjustments.length; ++i) {
+            uint64 scalingFactor = _getCurrentScalingFactor(
+                operator,
+                _hashStrategyAndOperatorSet(strategy, magnitudeAdjustments[i].operatorSet)
+            );
+            uint64 normalizedMagnitudeIncrement = SlashingAccountingUtils.normalizeMagnitude({
+                nonNormalizedMagnitude: magnitudeAdjustments[i].magnitudeDiff,
```

Comment:
i feel like they should send the magnitude rather than the nonnormalized magnitude and we should convert on the fly

Reply:
Was thinking the exact same thing looking at it. Allocators shouldn't have to think about the normalized magnitudes